python/cuml/common/base.pyx

```diff
@@ -152,3 +207,31 @@ class Base:
     def __setstate__(self, state):
         self.__dict__.update(state)
         self.handle = cuml.common.handle.Handle()
+
+    def __getattr__(self, attr):
+        real_name = '_' + attr
+        if real_name in self.__dict__.keys():
```

Comment:
minor nitpick but I always think hasattr() is more pythonic.. there are various schools of thought of the best way, but I think this anti __dict__ sentiment is common: https://stackoverflow.com/questions/9748678/which-is-the-best-way-to-check-for-the-existence-of-an-attribute